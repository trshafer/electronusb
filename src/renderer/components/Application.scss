.patient-details {
    display: flex;
}

.patient-details > * {
    flex: 1;
}
