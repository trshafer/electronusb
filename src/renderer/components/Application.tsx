import { hot } from 'react-hot-loader/root';
import * as React from 'react';

import { PatientContainer } from '../containers/PatientContainer';

const Application = () => (
    <div>
        Hello World from Electron!
        <PatientContainer />
    </div>
);

export default hot(Application);
