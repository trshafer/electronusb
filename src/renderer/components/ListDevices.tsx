import Table from 'react-bootstrap/Table';
import * as React from 'react';

import * as USB from 'usb';

interface ListProps {
    readonly devices: ReadonlyArray<Readonly<USB.Device>>;
}

const createUSBDeviceRow: React.FunctionComponent<USB.Device> = (device, index) => (
    <tr key={index}>
        <td>{device.busNumber}</td>
        <td>{device.deviceAddress}</td>
        <td>{device.portNumbers}</td>
        <td>{JSON.stringify(device.deviceDescriptor)}</td>
    </tr>
);

export const ListDevices: React.FunctionComponent<ListProps> = ({ devices }) => (
    <div>
        <h2>USB Devices</h2>
        <Table bordered={true} hover={true}>
            <thead>
                <tr>
                    <th>BusNumber</th>
                    <th>DeviceAddress</th>
                    <th>PortNumbers</th>
                    <th>DeviceDescriptor</th>
                </tr>
            </thead>
            <tbody>{devices.map(createUSBDeviceRow, this)}</tbody>
        </Table>
    </div>
);
