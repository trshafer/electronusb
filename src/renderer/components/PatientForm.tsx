import * as React from 'react';

require('./PatientForm.scss');

export interface Props {
    readonly name: string;
    readonly weight: number;

    readonly updateName: (name: string) => any;
    readonly updateWeight: (weight: number) => any;
}

export class PatientForm extends React.Component<Props> {
    readonly updateName = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.props.updateName(event.target.value);
    };

    readonly updateWeight = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.props.updateWeight(Number(event.target.value));
    };

    render() {
        return (
            <div className="patient-form">
                <div>Patient Form</div>
                <div>
                    <label>Name: </label>
                    <input type="text" value={this.props.name} onChange={this.updateName} />
                </div>
                <div>
                    <label>Weight: </label>
                    <input type="number" value={this.props.weight} onChange={this.updateWeight} />
                </div>
            </div>
        );
    }
}
