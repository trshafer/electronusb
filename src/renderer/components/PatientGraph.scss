.patient-graph {
    background-color: yellow;
}
