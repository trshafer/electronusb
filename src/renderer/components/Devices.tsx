import * as React from 'react';

import * as USB from 'usb';
import { USBDevices } from './ListDevices';

interface Props {
    readonly hasFetched: boolean;
    readonly devices: ReadonlyArray<Readonly<USB.Device>>;
    readonly updateDevices: (devices: ReadonlyArray<Readonly<USB.Device>>) => void;
}

export class Devices extends React.Component<Props> {
    private readonly updateUSB = () => {
        this.props.updateDevices(USB.getDeviceList());
    };

    componentDidMount() {
        if (!this.props.hasFetched) {
            console.log('fetching devices');
            this.updateUSB();
            USB.on('attach', this.updateUSB);
            USB.on('detach', this.updateUSB);
        }
    }

    componentWillUnmount() {
        USB.removeListener('attach', this.updateUSB);
        USB.removeListener('detach', this.updateUSB);
    }

    render() {
        return (
            <div>
                <USBDevices devices={this.props.devices} />
            </div>
        );
    }
}

// export const Devices: React.FunctionComponent<Props> = ({ devices, updateDevices }) => (
// );
