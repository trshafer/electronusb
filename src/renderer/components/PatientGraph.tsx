import * as React from 'react';

require('./PatientGraph.scss');

export interface Props {
    readonly name: string;
    readonly weight: number;
}

export const PatientGraph: React.FunctionComponent<Props> = ({ name, weight }) => (
    <div className="patient-graph">
        <div>This is where the graph would go.</div>
        <div>Name: {name}</div>
        <div>Weight: {weight}</div>
    </div>
);
