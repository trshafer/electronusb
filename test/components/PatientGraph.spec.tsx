import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { PatientGraph } from '../../src/renderer/components/PatientGraph';

describe('PatientGraph component', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<PatientGraph name={'test'} weight={100} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
