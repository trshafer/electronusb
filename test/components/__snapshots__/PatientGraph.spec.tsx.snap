// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientGraph component renders correctly 1`] = `
<div
  className="patient-graph"
>
  <div>
    This is where the graph would go.
  </div>
  <div>
    Name:
    test
  </div>
  <div>
    Weight:
    100
  </div>
</div>
`;
